components/brave_wallet/browser/simple_hash_client_unittest.cc

```diff
@@ -658,10 +660,27 @@ TEST_F(SimpleHashClientUnitTest, ParseNFTsFromSimpleHash) {
   })";
   json_value = base::JSONReader::Read(json);
   ASSERT_TRUE(json_value);
-  result = simple_hash_client_->ParseNFTsFromSimpleHash(*json_value,
-                                                        mojom::CoinType::SOL);
+  result = simple_hash_client_->ParseNFTsFromSimpleHash(
+      *json_value, mojom::CoinType::SOL, true, false);
   ASSERT_TRUE(result);
   EXPECT_EQ(result->second.size(), 0u);
+
+  // When only_spam is set and skip_spam is not, spam should be parsed
+  result = simple_hash_client_->ParseNFTsFromSimpleHash(
+      *json_value, mojom::CoinType::SOL, false, true);
+  ASSERT_TRUE(result);
+  EXPECT_EQ(result->second.size(), 1u);
+
+  // When only_spam is set and skip_spam is set, parsing should fail
+  result = simple_hash_client_->ParseNFTsFromSimpleHash(
+      *json_value, mojom::CoinType::SOL, true, true);
+  ASSERT_FALSE(result);
+
+  // When only_spam is false and skip_spam is false, spam should be parsed
+  result = simple_hash_client_->ParseNFTsFromSimpleHash(
+      *json_value, mojom::CoinType::SOL, false, false);
+  ASSERT_TRUE(result);
+  EXPECT_EQ(result->second.size(), 1u);
```

Comment:
the test json should at least have additional non-spam entry to show the different between (false, true) and (false, false)